integration/spark/app/src/main/java/io/openlineage/spark/agent/ArgumentParser.java

```diff
@@ -53,8 +53,6 @@ public class ArgumentParser {
       "spark.openlineage.jobName.replaceDotWithUnderscore";
   private static final String SPARK_CONF_FACETS_DISABLED = "spark.openlineage.facets.disabled";
 
-  private static final String SPARK_CONF_DEBUG_FACET = "spark.openlineage.debugFacet";
```

Comment:
We replace `spark.openlineage.debugFacet` `String` property with `spark.openlineage.facets.debug.enabled` boolean property

Reply:
Whilst I am all for unifying how we configure facets, there are several things that are missing for me:

1. Deprecation notices before removal
2. Retaining backwards compatibility whilst the deprecation notice is still in place.
3. Documentation changea about these changes

Reply:
Ad 3. Documentation will be changed
Ad 2 and 3. I'm not sure if maintaining backward compatibility and adding deprecation warnings is worth it. For `spark_unknown` and `spark.logicalPlan` we can't maintain it, because this pull request changes the how they are activated. We could do something about `debug` facet but is it really worth it? It would complicate the logic quite significantly (parsing the field, but hiding it; using it when present; deciding what takes precedence; how to handle merges; writing extra tests). If I add the warning, how long should it stay there? Maybe we could add warning (or even error) when we see the `spark.openlineage.debugFacet` configuration but don't use the value? But when should we remove it? These three facets are pretty advanced and used only when something is not working. Why don't we assume that when someone needs to debug the application they copy the value from the documentation anyway?

Reply:
I think for `debugFacet` specifically we could make the change outright - as the name suggests, it's the debug facet. There's no guarantee of stability of it and I really hope no one is using it in production.

Other setting

```
spark.openlineage.facets.disabled
```

we have to keep  - but the logic should not override 

```
spark.openlineage.facets.XXX.disabled
```

By that I mean 
```
spark.openlineage.facets.spark_unknown.disabled
spark.openlineage.facets.spark.logicalPlan.disabled
```
should be disabled, and lack of them in `spark.openlineage.facets.disabled` should not cause those two to be enabled unless someone adds the new-form config.

However, if someone added something additional to `spark.openlineage.facets.disabled` - it should be disabled.

So, basically, both configs, but new form having priority over the others, with sensible defaults - `spark_unknown`, `spark.logicalPlan` and `debug` disabled by default in the new form.

Reply:
As for documentation, deprecation warnings - definitely agree with Damien.